include/hse/experimental.h

```diff
@@ -0,0 +1,198 @@
+ */
+
+/* The interfaces defined in this file are for internal use only.  The are not
+ * part of the HSE library public API and hence may change in name, arguments,
+ * and/or semantics without notice.
+ */
+#ifndef HSE_EXPERIMENTAL_H
+#define HSE_EXPERIMENTAL_H
+
+#include <hse/flags.h>
+#include <hse/limits.h>
+#include <hse/types.h>
+#include <hse/version.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#pragma GCC visibility push(default)
+
+/* hse_kvdb_compact() flags */
+#define HSE_KVDB_COMPACT_CANCEL   (1u << 0)
+#define HSE_KVDB_COMPACT_SAMP_LWM (1u << 1)
+
+/** @addtogroup KVDB
+ * @{
+ */
+
+/** @brief Status of a compaction request. */
+struct hse_kvdb_compact_status {
+    unsigned int kvcs_samp_lwm;  /**< space amp low water mark (%). */
+    unsigned int kvcs_samp_hwm;  /**< space amp high water mark (%). */
+    unsigned int kvcs_samp_curr; /**< current space amp (%). */
+    unsigned int kvcs_active;    /**< is an externally requested compaction underway. */
+    unsigned int kvcs_canceled;  /**< was an externally requested compaction canceled. */
+};
+
+/** @brief Storage information for a KVDB. */
+struct hse_kvdb_storage_info {
+    uint64_t total_bytes;     /**< total space in the file-system containing this kvdb */
+    uint64_t available_bytes; /**< available space in the file-system containing this kvdb */
+    uint64_t allocated_bytes; /**< allocated storage space for a kvdb */
+    uint64_t used_bytes;      /**< used storage space for a kvdb */
+};
+
+/**@} KVDB */
+
+/** @addtogroup KVS
+ * @{
+ */
+
+/** @brief Number of keys found from a prefix probe operation. */
+enum hse_kvs_pfx_probe_cnt {
+    HSE_KVS_PFX_FOUND_ZERO = 0, /**< Zero keys found with prefix. */
+    HSE_KVS_PFX_FOUND_ONE,      /**< One key found with prefix. */
+    HSE_KVS_PFX_FOUND_MUL,      /**< Multiple keys found with prefix. */
+};
+
+/**@} KVS */
+
+/** @addtogroup KVDB Key-Value Database (KVDB)
+ * @{
+ */
+
+/** @brief Request a data compaction operation.
+ *
+ * In managing the data within an HSE KVDB, there are maintenance activities
+ * that occur as background processing. The application may be aware that it is
+ * advantageous to do enough maintenance now for the database to be as compact
+ * as it ever would be in normal operation.
+ *
+ * See the function hse_kvdb_compact_status_get().
+ *
+ * <b>Flags:</b>
+ * @arg HSE_KVDB_COMPACT_CANCEL - Cancel an ongoing compaction request.
+ * @arg HSE_KVDB_COMPACT_SAMP_LWM - Compact to the space amp low watermark.
+ *
+ * @note This function is thread safe.
+ *
+ * @param kvdb: KVDB handle from hse_kvdb_open().
+ * @param flags: Compaction flags.
+ *
+ * @remark @p kvdb must not be NULL.
+ * @remark @p flags must not be 0.
+ *
+ * @returns Error status.
+ */
+hse_err_t
+hse_kvdb_compact(struct hse_kvdb *kvdb, unsigned int flags);
```

Comment:
I would say just have the one KVDB doxygen section (ie merge the flags, functions, and types)